### PR TITLE
CI & Bug Fixes: 1D & 2D Compile

### PR DESCRIPTION
**Pull request by ax3l**
Add a modern CI entry that compiles 1D and 2D with `-Werror` and many warnings enabled.

### CODE REVIEW

**Review by prkkumar — APPROVED** [review]

**Comment by ax3l** [comment]
Something still defaults in 2D, let's test locally

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -49,6 +49,53 @@ jobs:
         cmake --build build_RZ -j 2
         ./build_RZ/bin/warpx Examples/Physics_applications/laser_acceleration/inputs_rz
 
+  build_2D_rz:
```
Comment (dpgrote) [inline .github/workflows/ubuntu.yml]: This all looks good. Though, this name should presumably be "build_1D_2D". For consistency, perhaps the name above should be "build_cxxminimal_3D_RZ"?

**Comment by dpgrote** [comment]
Maybe I missed the discussion, but presumably gcc12 is being used because it is better at doing the checks. Is there a reason that gcc12 is also not used for build_cxxminimal?

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (ax3l) [reply .github/workflows/ubuntu.yml]: Good point :)

**Comment by ax3l** [comment]
The reason I use a newer one here is to have a broad coverage in compilers we test. We should update some of the other tests to use it too, though. Will review that separately.

**Comment by ax3l** [comment]
> Something still defaults in 2D, let's test locally

Hm, I think EB in 1D and 2D (and likely RZ, too) is still segfaulting - even it its just enabled and not used.

To be followed up on Tue. #3686

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (ax3l) [reply .github/workflows/ubuntu.yml]: (not minimal because minimal is with respect to third-party dependencies)

**Pull request merged**